scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -1607,7 +1622,7 @@ public DtlsConnectorConfig build() {
 				}
 			} 
 
-			if (!config.clientAuthenticationRequired && !config.clientAuthenticationWanted
+			if (!clientOnly && !config.clientAuthenticationRequired && !config.clientAuthenticationWanted
```

Comment:
I think the idea is : if I'm a server and didn't ask for  any form of client authentication then I should not have set a trustcertificatetypes ?

!clientOnly does not mean server only and in case of "role exchange" this could be possible to have `trustCertificateTypes` used for client role and does not ask for client authentication in server role, but I concede this would be a pretty strange use case ... so maybe it's ok like this !

Reply:
> I think the idea is : if I'm a server and didn't ask for any form of client authentication then I should not have set a trustcertificatetypes ?

Yes.

> !clientOnly does not mean server only and in case of "role exchange" this could be possible to have trustCertificateTypes used for client role and does not ask for client authentication in server role, but I concede this would be a pretty strange use case ... so maybe it's ok like this !

Though it was more a idea, to check the configuration more strict, because it got very complex, it may also fail for use-case I didn't see. So check only for "server only" :-)-

Reply:
Just to mention:
if find some cases, where the validation is too strict, just report or fix it.